dbt_platform_helper/templates/pipelines/codebase/overrides/stack.ts

```diff
@@ -54,9 +54,17 @@ export class TransformedStack extends cdk.Stack {
         );
 
         for (const branch of watchedBranches) {
+            let sanitisedBranch = branch;
+
+            if (branch?.endsWith('*')){
+                sanitisedBranch?.replace('*', '.*');
+            } else {
+                sanitisedBranch = sanitisedBranch?.concat('$');
+            }
```

Comment:
@acodeninja I think this is enough, but I also recall our conversation around the branch having to allow for `feat-whatever` and `feat` and I think this is what covers it